applications/annotate_points.md

```diff
@@ -0,0 +1,383 @@
+            new_label = labels[new_ind]
+            current_properties['label'] = np.array([new_label])
+            points_layer.current_properties = current_properties
+
+
+        @viewer.bind_key('Control-S')
+        def save_points(event):
+            """Save the added points to a CSV file"""
+            # get the frame indices
+            frame_indices = np.unique(points_layer.data[:, 0]).astype(np.int)
+
+            # get the filenames
+            all_files = np.asarray(glob.glob(im_path))
+            file_names = all_files[frame_indices]
+
+            # create and write dataframe
+            header = pd.MultiIndex.from_product(
+                [[scorer], labels, ['x', 'y']],
+                names=['scorer', 'bodyparts', 'coords']
+            )
+            df = pd.DataFrame(
+                index=file_names,
+                columns=header,
+            )
+
+            # populate the dataframe
+            for label, coord in zip(points_layer.properties['label'], points_layer.data):
+                fname = all_files[coord[0].astype(np.int)]
+                df.loc[fname][scorer][label]['x'] = coord[2]
+                df.loc[fname][scorer][label]['y'] = coord[1]
+
+            # write the dataframe
+            df.to_csv(output_path)
+        
+```
+## point_annotator()
+We will create the GUI within a function called `point_annotator()`. Wrapping the GUI creation in the function allows us to  See below for the function definition.
+
+```python
+def point_annotator(
+	im_path: str,
+	labels: List[str],
+	output_path: str='anno.csv',
+	scorer: str='user'
+):
+    """Create a GUI for annotating points in a series of images.
+
+    Parameters
+    ----------
+    im_path : str
+        glob-like string for the images to be labeled.
+    labels : List[str]
+        list of the labels for each keypoint to be annotated (e.g., the body parts to be labeled).
+    output_path : str
+        path to the csv file to which the annotations will be saved
+    scorer : str
+        name of the person performing the annotation
+    """
+
+```
+
+## Loading the video
+First, we load the movie to be annotated. Since behavior movies can be quite long, we will use a lazy loading strategy (i.e., we will only load the frames as they are used). Using [`dask-image`](https://github.com/dask/dask-image), we can construct an object that we can pass to napari for lazy loading in just one line. For more explanation on using dask to lazily load images in napari, see [this](https://napari.org/tutorials/applications/dask) tutorial.
+
+```python
+stack = imread(im_path)
+```
+
+We can then start the viewer Note that we use the `napari.gui_qt()` context manager to start and manage Qt event loop. We must provide the `contrast_limits` and set `is_pyramid=False` to prevent the image stack from being evaluated (i.e., loaded) upon layer creation. All of the following GUI code should be within the `napari.gui_qt()` context manager.
```

Comment:
This kwarg is now `is_multiscale`, but that shouldn't actually be needed anymore and isn't in your function below. Should drop line

Reply:
Fixed!